pkg/controllers/user/cis/clusterScanHandler.go

```diff
@@ -111,40 +109,24 @@ func (csh *cisScanHandler) Create(cs *v3.ClusterScan) (runtime.Object, error) {
 	if !v3.ClusterScanConditionCreated.IsTrue(cs) {
 		logrus.Infof("cisScanHandler: Create: deploying helm chart")
 		currentK8sVersion := cluster.Spec.RancherKubernetesEngineConfig.Version
-		shortK8sVersion := util.GetTagMajorVersion(currentK8sVersion)
-		cisConfigParams, err := kontainerdrivermetadata.GetCisConfigParams(
-			shortK8sVersion,
-			csh.cisConfigLister,
-			csh.cisConfigClient,
-		)
-		if err != nil {
-			logrus.Debugf("cisScanHandler: Create: benchmark version not found for k8s version: %v(%v), using default",
-				currentK8sVersion, shortK8sVersion)
-			cisConfigParams, err = kontainerdrivermetadata.GetCisConfigParams(
-				"default",
-				csh.cisConfigLister,
-				csh.cisConfigClient,
-			)
-			if err != nil {
-				return cs, fmt.Errorf("error fetching default cis config: %v", err)
-			}
+		overrideBenchmarkVersion := ""
+		if cs.Spec.ScanConfig.CisScanConfig != nil {
+			overrideBenchmarkVersion = cs.Spec.ScanConfig.CisScanConfig.OverrideBenchmarkVersion
 		}
-		benchmarkInfo, err := kontainerdrivermetadata.GetCisBenchmarkVersionInfo(
-			cisConfigParams.BenchmarkVersion,
-			csh.cisBenchmarkVersionLister,
-			csh.cisBenchmarkVersionClient,
+		bv, bvManaged, err := GetBenchmarkVersionToUse(overrideBenchmarkVersion, currentK8sVersion,
+			csh.cisConfigLister, csh.cisConfigClient,
+			csh.cisBenchmarkVersionLister, csh.cisBenchmarkVersionClient,
 		)
 		if err != nil {
-			return nil, fmt.Errorf("cisScanHandler: Create: error fetching benchmark version info %v: %v",
-				cisConfigParams.BenchmarkVersion, err)
+			return cs, err
```

Comment:
When version comparison fails, does this keep returning error? Is there any gated logic to prevent invalid cluster scan being re-enqueued into controller?

Reply:
It's gated at the create time itself to reject the scan.

Reply:
actions_cis.go

Reply:
This means we will never remove any benchmarkversion data from kdm - i that correct @leodotcloud? because if we change kdm, there will be clusterscans that might become invalid and get re-enqueued?

Reply:
Thus even the "MinKubernetesVersion" in the"default" entry in kdm should never change?

Reply:
Correct, we will ever remove any benchmark data versions from kdm. Typically anything in kdm, once it's released, we can never remove it. 

But I cannot guarantee that the min k8s versions would never change, but most likely not.

Reply:
As discussed offline, older k8s versions will pick the "default" entry and use the corresponding benchmark version which will have a min k8s version which is 1.15. The default is mainly to allow newer k8s versions to use the latest available benchmark version. The current PR enforces the min k8s version so that older k8s versions don't allow the scan.